backend/src/main/kotlin/vegait/rs/osipodgorica/service/CategoryService.kt

```diff
@@ -19,11 +25,42 @@ class CategoryService(val categoryRepo: CategoryRepository, val uploadService: I
         return categoryRepo.save(category)
     }
 
+    fun update(request: UpdateCategoryRequest, id: Long) =
+        runCatching {
+            categoryRepo.findById(id).get()
+        }.mapCatching { oldCategory ->
+            Category(
+                id = id,
+                name = request.name,
+                relativeUrl = request.thumbnail?.storeNewIcon(id, oldCategory.relativeUrl)
+                    ?: oldCategory.relativeUrl,
+            )
+        }.mapCatching { category ->
+            categoryRepo.save(category)
+        }.getOrThrow()
+
+    private fun MultipartFile.storeNewIcon(id: Long, oldIconPath: String?): String? =
+        runCatching {
+            oldIconPath?.let { scopedOldIconPath ->
+                imageUploadService.deleteFile(scopedOldIconPath)
+            }
+        }.mapCatching {
+            imageUploadService.store(this, "categories/$id")
+        }.getOrNull()
+
     fun get(id: Long): Category {
         return categoryRepo.findById(id).orElseThrow()
     }
 
     fun index(): List<Category> {
         return categoryRepo.findAll()
     }
-}
\ No newline at end of file
+
+    fun delete(id: Long) {
```

Comment:
Ako dođe do greške prilikom brisanja foldera i fajlova, hoće li biti vraćeni podaci o kategoriji? Jesmo li sigurni da ne možemo obrisati kategoriju ako ima lokacija vezanih za nju?

Reply:
Kad to probam da uradim na postman-u odbija mi. Samo mogu da izbrisem kategorije koje su nezavistne od lokacije

Ovo je error:
`FOREIGN KEY (`category_id`) REFERENCES `categories``